db/db.go

```diff
@@ -289,3 +289,30 @@ func (db *pgDB) CountOffchainData(ctx context.Context) (uint64, error) {
 
 	return count, nil
 }
+
+// DetectOffchainDataGaps returns the number of gaps in the offchain_data table
+func (db *pgDB) DetectOffchainDataGaps(ctx context.Context) (map[uint64]uint64, error) {
+	const detectBatchNumGapQuery = "SELECT * FROM vw_batch_num_gaps;"
+
+	rows, err := db.pg.QueryxContext(ctx, detectBatchNumGapQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	gaps := make(map[uint64]uint64)
+	for rows.Next() {
+		data := struct {
+			CurrentBatchNum uint64 `db:"current_batch_num"`
+			NextBatchNum    uint64 `db:"next_batch_num"`
+		}{}
```

Comment:
WDYT if the anonymous `struct` is declared only once, namely before the `for` loop and have a reference of the struct type within each iteration? It might be somewhat more performant (though chances are that Golang compiler knows how to optimize this as well and extract it outside of the for loop anyways)...?

Reply:
Agree, done